### PR TITLE
Packages: Move `migration` to new structure

### LINKED ISSUE #6175: Packages: Move `migration` to new structure
## Feature Description

* [ ] Move code from `/assets/src/migration` to `/packages/i18n/migration/`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `/packages/migration/package.json` referencing its dependencies
* [ ] Create `/packages/migration/README.md` with short documentation.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Move files to new structure.

## Relevant Technical Choices

Create a bin file that can be used in cli. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6175

### CODE REVIEW

File: assets/src/dashboard/app/serializers/savedTemplates.js [hunk 1]
```diff
@@ -20,9 +20,9 @@
 import { __ } from '@web-stories-wp/i18n';
 
 /**
- * Internal dependencies
+ * External dependencies
```
Comment (swissspidy) [inline assets/src/dashboard/app/serializers/savedTemplates.js]: There's already a `external dependencies` section above

File: packages/migration/src/migrations/v0017_inlineTextProperties.js [hunk 2]
```diff
@@ -128,8 +128,6 @@ function addInlineLetterSpacing(content, letterSpacing) {
   return wrapWithSpan(content, `letter-spacing: ${letterSpacing / 100}em`);
 }
 
-/* eslint-disable security/detect-non-literal-regexp */
```
Comment (swissspidy) [inline packages/migration/src/migrations/v0017_inlineTextProperties.js:131]: Why was this removed? This file shouldn't be changed.

**Comment by swissspidy** [comment]
The file permissions of `packages/migration/src/index.js` should stay 644, not change to 755:

Screenshot 2021-02-08 at 13 33 35

File: packages/migration/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "main": "./src/index.js",
+  "bin": {
+    "migration": "./src/cli.js"
+  },
+  "dependencies": {
+    "uuid": "^8.3.2",
+    "polished": "^4.0.5"
+  },
+  "devDependencies": {
+    "@rollup/plugin-node-resolve": "^11.1.1"
```
Comment (swissspidy) [inline packages/migration/package.json:22]: `"rollup": "^2.36.2",` should also be listed here, and removed from `package.json`

**Review by swissspidy — CHANGES_REQUESTED** [review]
Two remaining nits about adding `rollup` to `package.json` and inadvertent file permission changes.

File: packages/migration/package.json (hunk 3 above)
Comment (spacedmonkey) [reply packages/migration/package.json:22]: Fixed.

File: packages/migration/src/migrations/v0017_inlineTextProperties.js (hunk 2 above)
Comment (spacedmonkey) [reply packages/migration/src/migrations/v0017_inlineTextProperties.js:131]: I don't know. Odd. Fixed now - https://github.com/google/web-stories-wp/pull/6244/commits/5d06de165ea669eb143cb3ab7fc115f11e47e922

**Review by swissspidy — APPROVED** [review]
